### PR TITLE
got simple darray test working, did some cleanup

### LINKED ISSUE #137: fix warnings in pioc.c
There are some warnings that should be fixed:

/home/ed/pio3/ParallelIO/src/clib/pioc.c: In function ‘PIOc_InitDecomp’:
/home/ed/pio3/ParallelIO/src/clib/pioc.c:255:55: warning: passing argument 3 of ‘subset_rearrange_create’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
         ierr = subset_rearrange_create( *ios, maplen, compmap, dims, ndims, iodesc);
                                                       ^~~~~~~
In file included from /home/ed/pio3/ParallelIO/src/clib/pioc.c:12:0:
/home/ed/pio3/ParallelIO/src/clib/pio_internal.h:153:9: note: expected ‘long long int *’ but argument is of type ‘const long long int *’
     int subset_rearrange_create(const iosystem_desc_t ios, int maplen, PIO_Offset compmap[],
         ^~~~~~~~~~~~~~~~~~~~~~~

### PR DESCRIPTION
**Pull request by edhartnett**
Got the simple darray test working. Did some cleanup in files I was looking through.

This is in pursuit of  #89. Also Fixes #137.

Except for test code, all changes to code are format and whitespace only. (And adding logging statements to help debug my test.)

### CODE REVIEW

**Comment by jayeshkrishna** [comment]
Mixing formatting+doc changes with code changes makes it harder to review changes. We should keep the code changes separate from doc/formatting changes.

File: src/clib/pio_lists.c [hunk 1]
```diff
@@ -251,22 +254,21 @@ int pio_add_to_iodesc_list(io_desc_t *iodesc)
...
         ciodesc = current_iodesc;
-    for(ciodesc = pio_iodesc_list; ciodesc != NULL; ciodesc = ciodesc->next)
-    {
-        if(ciodesc->ioid == abs(ioid))
-        {
-            current_iodesc = ciodesc;
-            break;
-        }
-    }
+    else
```
Comment (jayeshkrishna) [inline src/clib/pio_lists.c]: Is this change (although required) related to this PR?

Comment (edhartnett) [inline src/clib/pio_lists.c]: Cleaning as I touch the code. We did our whitespace sweep - any other whtiespace issues should be addressed as found. Eventually they will decrease to a smaller and smaller number.

**Comment by rljacob** [comment]
In our telecon, we definitely agreed to not mix clean-up with other development on a branch/PR.

**Pull request closed without merging**